pulsar-client/src/main/java/org/apache/pulsar/client/impl/transaction/TransactionImpl.java

```diff
@@ -63,11 +61,18 @@
 
     private final ArrayList<CompletableFuture<MessageId>> sendFutureList;
     private final ArrayList<CompletableFuture<Void>> ackFutureList;
+    private volatile State state;
+
+    public enum State {
+        OPEN,
+        CLOSE
```

Comment:
Why not keep the state consistent with the transaction state? Open/Committing/Aborting/Committed/Aborted ?

Reply:
some situation we can't get the real state fro this transaction, example: the transaction timeout, it will be aborted but client commit.

Reply:
So the state should be committing -> closed ?

Reply:
commit or abort finish, the state will become closed. we don't need to change the state to committing or aborting, it seem not useful. When we need committing or aborting sate we can add.